### PR TITLE
feat(spur-cli): support slurm-buildkite bridge compatibility

### LINKED ISSUE #434: Spur features needed for CliMA/slurm-buildkite (Buildkite CI) to work
We want to run our Buildkite CI on Spur using the standard bridge, [CliMA/slurm-buildkite](https://github.com/CliMA/slurm-buildkite) (a poller submits one `sbatch` job per queued Buildkite job, which then runs `buildkite-agent start --acquire-job <uuid>`).

Its stock scheduler talks plain Slurm CLI. The following features are missing on Spur and are needed for it to work. Reproduced with real jobs on real compute nodes on **both Spur 0.4.1 and 0.3.0** — none of these were addressed in the 0.4.0/0.4.1 releases.

### Features needed on Spur

- [ ] **`--comment` readback** — `sbatch --comment=<url>` must be retrievable afterwards (via `squeue %k`, `scontrol show job`, or `sacct`). This is how `slurm-buildkite`'s **stock `SlurmJobScheduler`** maps a Buildkite job to its Spur job id — statelessly, with no local DB. (A stateful scheduler *could* instead keep a local url→jobid map and skip `--comment` — but the framework's autodetect sees `sbatch`/`sinfo`, selects the **Slurm** path for Spur, and that path requires it.) Currently the comment is accepted but unreadable by any command. Verified on a live running job: submitted with `--comment=COMMENT_SENTINEL_9F3A7`, the sentinel appears in **none** of `squeue -o %k` (prints `?`), `scontrol show job` (no `Comment=` field at all), `spur show job`, or `sacct`.
- [ ] **`sbatch --parsable`** — print the bare integer job id. Currently: `Error: unexpected argument '--parsable' found`.
- [ ] **Forward positional script args** — `sbatch script.sh <arg>` must pass `<arg>` to the script as `$1`. Currently: `Error: unexpected argument '<arg>' found`.
- [ ] **`squeue --format` with literals + `%k`/`%A`** — e.g. `--format=%k,%A` must produce `comment,jobid` so it can be split on `,`. Currently `%k`/`%A` render as `?` and literal characters are dropped.
- [ ] **`squeue --name=<name>`** — filter by job name. Currently: `Error: unexpected argument '--name' found`.

`scancel --name=<name>` already works and needs no change.

### The three commands the bridge runs

```
sbatch  --parsable --job-name=buildkite --comment=<build_url> --output=slurm-%j.log [slurm_* flags] schedule_job.sh <uuid>
squeue  --name=buildkite --noheader --format=%k,%A
scancel --name=buildkite <job_ids...>
```

Only `scancel` works today. With the five features above, CliMA/slurm-buildkite runs on Spur unmodified.

Happy to send a Spur PR for these if welcome.

### PR DESCRIPTION
**Pull request by shiv-tyagi**
## Summary

Adds the missing Slurm CLI features required by [CliMA/slurm-buildkite](https://github.com/CliMA/slurm-buildkite) and similar CI bridges that script against Slurm's CLI surface.

- **`sbatch --parsable`**: prints bare job ID on stdout (no header/decoration)
- **`sbatch` positional script args**: trailing arguments after the script path are forwarded as `$1`, `$2`, etc. via `set --` injection
- **`--comment` readback**: new `comment` field on `JobInfo` proto, surfaced in `squeue %k` and `scontrol show`
- **`squeue --format` literals**: refactored format engine to preserve literal characters between specifiers (`%k,%A` emits `comment,42` with no extra spacing), added `%%` escape support
- **`squeue %A`**: resolves to per-element job ID (matches Slurm semantics)
- **`squeue --name`**: client-side filter by job name

## Approach

- Proto: `script_args` (field 81) on `JobSpec`, `comment` (field 53) on `JobInfo`. Both default to empty, backward-compatible with existing Raft state via `#[serde(default)]`.
- Format engine: introduced `FormatToken` enum (`Field | Literal`) replacing the old `Vec<FormatField>` + `join(" ")` pattern. All callers (`squeue`, `sinfo`, `sacct`) updated. Named-format path (`sacct`) emits literal space separators between fields.
- Agent: `build_job_script` injects `set -- <args>` after the shebang line (or at top if no shebang) when script args are present. Handles CRLF shebangs and shell-escapes args via `shlex`.

## Test plan

- [x] 123 spur-cli unit tests pass (format engine, sbatch, squeue, sacct)
- [x] spurd `build_job_script` / `inject_script_args` tests (shebang, no-shebang, CRLF, quoting)
- [x] Full workspace `cargo test` and `cargo clippy` clean
- [x] Smoke tested on 4-node bare-metal cluster: `sbatch --parsable`, `--comment` readback via `squeue %k,%A`, `scontrol show`, `squeue --name`, positional script args

Closes #434

### CODE REVIEW

File: crates/spur-cli/src/squeue.rs [hunk 1]
```diff
@@ -155,6 +165,8 @@ fn resolve_job_field(job: &spur_proto::proto::JobInfo, spec: char) -> String {
         'V' => format_timestamp(job.submit_time.as_ref()),
         'v' => job.reservation.clone(),
         'e' => format_timestamp(job.end_time.as_ref()),
+        'k' => job.comment.clone(),
+        'A' => job.job_id.to_string(),
```
[the remark this answers is not in the document]
Comment (shiv-tyagi) [reply crates/spur-cli/src/squeue.rs:164]: This is intentional. Slurm defines `%A` as the per-element job ID ("Job id. This will have a unique value for each element of job arrays"), not the array parent. `%F` is the base/parent ID. The previous code had this backwards. Verified against the Slurm man page and on a live cluster with array jobs.

**Review by yansun1996 — COMMENTED** [review]
## Summary

This is a well-scoped, well-tested PR — the format-engine literal-preservation redesign in particular is nicely done. A couple of minor things worth a look before merging.

## What's good

- **`format_engine.rs`'s `FormatField` → `FormatToken` refactor** is the right way to solve "buildkite needs `%k,%A` to emit `comment,42` with a literal comma, not a space-joined `comment , 42`" — preserving literal runs between specifiers rather than always space-joining fields. Genuinely nice: `test_default_format_spacing_unchanged` explicitly proves the *existing* default squeue/sinfo format output is byte-identical to before, which is exactly the backward-compat check I'd have asked for.
- **`script_args` / `set --` injection** (`agent_server.rs`) uses `shlex::try_join` to shell-escape each argument before injecting `set -- <args>` after the shebang — correctly prevents a malicious/malformed script argument from escaping into shell metacharacter territory. Good edge-case test coverage (shebang, no shebang, `#!/usr/bin/env bash`, CRLF shebang).
- **`candidate_script_path`'s rewrite to a real clap pre-parse** (instead of the old "assume the last positional arg" heuristic) is more robust, and — importantly — `script_args` is declared `trailing_var_arg + allow_hyphen_values`, which correctly avoids the exact class of bug found and fixed elsewhere in this repo recently (a flag like `-V` being misinterpreted as sbatch's own flag instead of being passed through to the wrapped script/command). Confirmed this is handled correctly here.
- The `scheduler_loop.rs`/`server.rs`/`accounting/grpc.rs`/`spur-k8s/job_controller.rs` touches are all legitimate, mechanical propagation of the two new fields (`script_args`, `comment`) through every `JobSpec`/`JobInfo` conversion site — not scope creep.
- `proto/slurm.proto` changes are clean appends (`script_args = 81`, `comment = 53`), no renumbering, `package slurm;` untouched.

## Two things worth a look

1. **`%A` (new `ARRAY_JOB_ID` format spec) resolves to `job.job_id`, not `job.array_job_id`.** (`squeue.rs`: `'A' => job.job_id.to_string()`.) For a non-array job these are the same, but for an individual task within a job array, real Slurm's `%A` means "the array's master job ID" — which should be `job.array_job_id` (already available on `JobInfo`, used a few lines away for the `array_job_id`/`array_task_id` proto fields), not the task's own ID. Worth a quick fix: `if job.array_job_id != 0 { job.array_job_id } else { job.job_id }` (or whatever this codebase's existing convention is for "0 means not an array job").

2. **`squeue --name`/`-n` filters client-side, unlike its sibling filters.** `--account`/`-A` is passed through as a field on `GetJobsRequest` (server-side filter), but `GetJobsRequest` has no `name` field, so `--name` ended up implemented as a post-fetch `.retain()`-style loop in the CLI. Functionally correct, but inconsistent with the existing pattern, and means `squeue -n foo` on a cluster with many jobs fetches and discards everything server-side would otherwise have filtered. Given this PR already touches `proto/slurm.proto` for two other fields, adding `string name = 6;` to `GetJobsRequest` and filtering server-side (matching `account`/`partition`/`user`) would be a small, consistent addition — or if there's a reason to keep it client-side for now, worth a one-line note on why.

Neither of these is a blocker — happy to see this land with a fast-follow, or fixed inline if it's a quick change. Nice work on the format-engine piece especially.

(Reviewed by direct code inspection rather than a full build/test run, given the PR's modest size and the focused nature of the changes — happy to do a build-level pass too if useful.)

**Comment by shiv-tyagi** [comment]
Thanks for the thorough review, Yan.

On your two points:

**[1] `%A` semantics** — This is correct as-is. Slurm's man page defines `%A` as "Job id. This will have a unique value for each element of job arrays" (the per-element ID), while `%F` is "Job array's job ID. This is the base job ID" (the parent). So `%A -> job_id` is right. Verified on a live cluster with array jobs: elements with `job_id` 5,6 and `array_job_id` 4 correctly show their own IDs via `%A`. The header label `ARRAY_JOB_ID` is admittedly confusing, but the resolver is correct.

**[2] `squeue --name` client-side filtering** — Good catch. Fixed in the latest push: added `string name = 6` to `GetJobsRequest`, wired server-side filtering through `cluster.get_jobs()` (comma-separated OR, matching the existing pattern for `user`/`partition`/`account`), updated the REST handler and `JobsQuery`, migrated `scancel --name` to server-side as well, and added a unit test for the name filter. All verified on the bare-metal cluster and via REST.

**Review by yansun1996 — APPROVED** [review]
Both findings from my earlier review are resolved — the `--name` filter is now server-side (matches the sibling `user`/`partition`/`account` filters), and my `%A` concern was a false positive on my part (verified: real Slurm's `%A` is the per-element job ID, not the array parent — that's `%F` — thanks for catching that with the man page + live-cluster verification). LGTM.
